src/export_from_inboveg/HT3260_data_preparation.Rmd

```diff
@@ -0,0 +1,420 @@
+
+```
+
+
+### combined
+= site_characteristics = structure + layer
+```{r}
+site_characteristics <- bind_rows(structure, layerqual) %>%
+    mutate(var = str_to_lower(var),
+                 value_class = str_to_lower(value_class))
+  
+```
+
+
+## vegetation
+rearrangement of vegetation
+```{r}
+vegetation <- vegetation_3260 %>% 
+  mutate(name = ifelse(is.na(name_scientific), name_original, name_scientific)) %>% 
+  select(survey, recording_givid, name, phenology_code, species_cover_code, species_cover, scale) %>% 
+  distinct() # tmp - to do opvolgen of dubbels eruit zijn
+
+# to do (later): nagaan of Nederlandse naam beschikbaar is in inboveg (@ Sofie of Pieter) 
+
+```
+
+controle - tmp
+```{r, eval=FALSE}
+# lijst my taxa 
+vegetation_3260 %>% 
+  filter(is.na(name_scientific)) %>% 
+  count(name_original) %>% 
+  arrange(name_original)
+
+sel_species <- c("Enteromorpha sp.")
+sel_species <- c("Salix 'alba' groep")
+sel_species <- c("Poaceae", "poaceae")
+
+vegetation_3260 %>% 
+  filter(name_original %in% sel_species) %>% 
+  left_join(select(header, recording_givid, user_reference), by = "recording_givid") %>% 
+  select(recording_givid, user_reference, name_original, species_cover_code) 
+  
+# soorten van 'my taxa' uniformiseren want identiek: to do (opvolgen; @ Pieter gevraagd)
+#   Callitriche 'obtusangula/platycarpa' groep en Callitriche platycarpa/obtusangula   -
+#   darmwier   en Enteromorpha sp.
+# gras (poaceae, vegetatief, niet gedetermineerd),   gras vegetatief en Poaceae, poaceae
+# mos en mos niet gedetermineerd
+# Salix 'alba' groep en Salix smalbladig en wilg smal
+
+```
+
+
+# data controle 
+##  input
+```{r, eval = FALSE}
+this_sample <- "Dommel_14"
+
+header_sel <- header_3260 %>% filter(str_detect(user_reference, this_sample))
+
+classif_sel <- classif_3260  %>% semi_join(header_sel, by = "recording_givid")
+structure_sel <- structure_3260  %>% semi_join(header_sel, by = "recording_givid")
+sitequal_sel <- sitequal_3260  %>% semi_join(header_sel, by = "recording_givid")
+layerqual_sel <- layerqual_3260  %>% semi_join(header_sel, by = "recording_givid")
+vegetation_sel <- vegetation_3260  %>% semi_join(header_sel, by = "recording_givid")
+
+# is ok; alles zit erin
+
+```
+
+##  output
+
+```{r, eval = FALSE}
+header_sel <- header %>%  
+  filter(str_detect(user_reference, this_sample))
+
+site_characteristics_sel <- site_characteristics %>% semi_join(header_sel, by = "recording_givid")
+vegetation_sel <- vegetation %>% semi_join(header_sel, by = "recording_givid")
+
+# is ok; alles zit erin
+
+```
+
+# data export
+
+```{r}
+write_vc(header, "HT3260_header", 
+         root = path, sorting = c("recording_givid"))
+write_vc(site_characteristics, "HT3260_site_characteristics", 
+         root = path, sorting = c("recording_givid", "var", "value_class"))
+write_vc(vegetation, "HT3260_vegetation", 
+         root = path, sorting = c("recording_givid", "name", "phenology_code"))
```

Comment:
Hier krijgen ik een warning dat deze sorteervariabelen niet tot een unieke sortering leidt. Dus meerdere records voor eenzelfde combinatie van sorteervariabelen.

Je ziet twee bedekking voor een zelfde combinatie soort/fenologie bij recording_givid = IV2017103112000035 als volgende code runt:

```suggestion
write_vc(vegetation, "HT3260_vegetation", 
         root = path, sorting = c("recording_givid", "name", "phenology_code"))
         
  vegetation %>%
    group_by(recording_givid,  name, phenology_code) %>%
    filter(n() > 1)
```

Reply:
klopt; dit moet nog aangepast worden in inboveg (zoals ook nog een aantal andere zaken aangepast moeten worden)